app/sync.js

```diff
@@ -189,11 +189,19 @@ module.exports.onSyncReady = (isFirstRun, e) => {
    * @param {Immutable.Map} site
    */
   const folderToObjectId = {}
-  const syncBookmark = (site) => {
-    if (!site || (site.get('objectId') && seed.equals(site.get('originalSeed'))) ||
-      folderToObjectId[site.get('folderId')] || !syncUtil.isSyncable('bookmark', site)) {
-      return
+  const shouldSyncBookmark = (site) => {
```

Comment:
minor enhancement: if you apply this as a filter function before sorting in https://github.com/brave/browser-laptop/pull/8409/files#diff-b7794baa4e0cd1fb8cab9449660662baR229, it might improve things slightly since then there are less things to sort